### PR TITLE
CMake: several code cleanups

### PR DESCRIPTION
**Pull request by DerDakon**
I've build curl successfully with 3.0.2 and 3.16.2 afterwards.

### CODE REVIEW

File: CMake/FindNSS.cmake [hunk 1]
```diff
@@ -10,6 +10,8 @@ set(NSS_LIBRARIES ${PC_NSS_LINK_LIBRARIES})
 set(NSS_INCLUDE_DIRS ${PC_NSS_INCLUDE_DIRS})
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(NSS DEFAULT_MSG NSS_INCLUDE_DIRS NSS_LIBRARIES)
+find_package_handle_standard_args(NSS
+    REQUIRED_VARS NSS_LIBRARIES NSS_INCLUDE_DIRS
+    VERSION_VAR PC_NSS_VERSION)
```
Comment (Lekensteyn) [inline CMake/FindNSS.cmake]: `PC_NSS_VERSION` will not exist on platforms without PkgConfig.
I think that this is ok since CMakeLists.txt does not specify a version requirement, making this purely cosmetic (right?)

**Comment by jay** [comment]
Maybe @jzakrzewski or @Lekensteyn will know? I'm fine with it as long as it doesn't bump the minimum version required.

File: CMakeLists.txt [hunk 2]
```diff
@@ -102,14 +102,19 @@ option(ENABLE_CURLDEBUG "Set to ON to build with TrackMemory feature enabled" OF
...
+    foreach(_CCOPT -pedantic -Wall -W -Wpointer-arith -Wwrite-strings -Wunused -Wshadow -Winline -Wnested-externs -Wmissing-declarations -Wmissing-prototypes -Wno-long-long -Wfloat-equal -Wno-multichar -Wsign-compare -Wundef -Wno-format-nonliteral -Wendif-labels -Wstrict-prototypes -Wdeclaration-after-statement -Wcast-align -Wtype-limits -Wold-style-declaration -Wmissing-parameter-type -Wempty-body -Wclobbered -Wignored-qualifiers -Wconversion -Wno-sign-conversion -Wvla -Wdouble-promotion -Wno-system-headers -Wno-pedantic-ms-format)
       # surprisingly, CHECK_C_COMPILER_FLAG needs a new variable to store each new
       # test result in.
       check_c_compiler_flag(${_CCOPT} OPT${_CCOPT})
```
Comment (Lekensteyn) [inline CMakeLists.txt]: Consider reverting this change and use something like:
```
string(MAKE_C_IDENTIFIER "OPT${_CCOPT}" _optvarname)
check_c_compiler_flag(${_CCOPS} "${_optvarname}")
```

**Comment by bagder** [comment]
This PR is currently at 13 commits but clearly some of the commits fixes earlier commits in the series itself, and we generally frown upon that. They should be reduced to a set of separate commits that each have their own commit message and reasoning.

If that's too much work, I would rather squash them into a single commit.

File: CMake/FindNSS.cmake (hunk 1 above)
Comment (DerDakon) [reply CMake/FindNSS.cmake]: correct, this will simply do nothing if the variable is not set.

File: CMakeLists.txt (hunk 2 above)
Comment (DerDakon) [reply CMakeLists.txt]: good idea, done

**Comment by Lekensteyn** [comment]
I think that all CMake changes could be squashed into a single commit, "CMake: cleanup and simplify" or similar. You can keep the current subjects as bullet points in the description if desired. How does that sound?

File: CMakeLists.txt [hunk 3]
```diff
@@ -38,7 +38,8 @@
 # To check:
 # (From Daniel Stenberg) The cmake build selected to run gcc with -fPIC on my box while the plain configure script did not.
 # (From Daniel Stenberg) The gcc command line use neither -g nor any -O options. As a developer, I also treasure our configure scripts's --enable-debug option that sets a long range of "picky" compiler options.
-cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.0...3.16 FATAL_ERROR)
```
Comment (jay) [inline CMakeLists.txt]: Why did you change this to put in a max version

Comment (DerDakon) [reply CMakeLists.txt]: It sets all policies introduced up to 3.16 to new, for all versions that support this syntax. Older versions will just ignore that.
https://cmake.org/cmake/help/v3.16/command/cmake_minimum_required.html

Comment (jay) [reply CMakeLists.txt]: I'm not sure if I understand this. Is it that if a later version for example 3.17 introduces some new policies then this warns about them?

Comment (DerDakon) [reply CMakeLists.txt]: Policies warn only if code is hit where a policy would make a difference in behavior _and_ the policy is not explicitely set to either OLD or NEW. So if there is code that would change behavior in 3.18 and is build with that version then there would be warnings. Since 3.16 is the latest released version I don't think it's a good idea to set this to any higher value. So far only 2 policy warnings were triggered (CMP0054 and CMP0075), so most others were just silent because no affected code was used.

Comment (jay) [reply CMakeLists.txt]: Ok. My concern was that it functions on all versions from 3.0 on, which I guess it does. As you can tell I don't know a lot about cmake.

Comment (DerDakon) [reply CMakeLists.txt]: Yes, this syntax has been chosen because older versions will silently ignore it.

**Comment by jay** [comment]
Thanks
